packages/instant-apps-components/src/components/instant-apps-create/instant-apps-create.test.ts

```diff
@@ -0,0 +1,123 @@
+                portal,
+            },
+            });
+    
+        //const inline = document.getElementById('create-inline');
+
+
+        create.portal = portal;
+        create.content = webscene;
+
+        document.body.appendChild(create);
+
+        await new Promise(resolve => requestIdleCallback(resolve));
+        await new Promise((resolve) => setTimeout(resolve, 5000));
+
+        const shadowRoot = document.getElementById('create-inline')?.shadowRoot;
+        
+        const listOptions  = shadowRoot?.querySelectorAll(".instant-apps-create__option");
+
+        expect(listOptions?.length).toBe(5);
+
+        const hrefs = listOptions && Array.from(listOptions).map((option: HTMLAnchorElement) => option.href);
+        const validSubstrings = ["instantgallery", "mapviewer", "stories", "template", "dashboards"];
+        
+        const testValidity = (testString: string, substrings: string[]) => 
+            substrings.some(substring => {
+                if (testString.includes(substring) && validSubstrings.includes(substring)){
+                    delete validSubstrings[substring]
+                    return true;
+                }
+                return false;
+            });
+        const hrefsAreValid = hrefs?.every(href => testValidity(href, validSubstrings));
+
+        expect(hrefsAreValid).toBe(true);
+
+
+
+    });
+});
+
+describe('popover', async () =>{
+    const create = document.createElement('instant-apps-create');
+    create.setAttribute('mode','popover');
+    create.setAttribute('show-scale-bar','true');
+    create.setAttribute('id','create-popover');
+
+    test('test if create option map works in popover', async () => {
+        config.portalUrl = 'https://nw-brews.mapsdevext.arcgis.com';
+        const portal = new Portal({
+                url: 'https://nw-brews.mapsdevext.arcgis.com', // dev test
+            });
+
+
+        const webscene = new WebScene({
+            portalItem: {
+                id: 'd8881fc5e49d4bbcbd62ed23a620b89e', // dev test - scene
+                portal,
+            },
+            });
+    
+        //const inline = document.getElementById('create-inline');
+
+
+        create.portal = portal;
+        create.content = webscene;
+
+        document.body.appendChild(create);
+
+        await new Promise(resolve => requestIdleCallback(resolve));
+        await new Promise((resolve) => setTimeout(resolve, 5000));
+
+        const shadowRoot = document.getElementById('create-popover')?.shadowRoot;
+
+        const listOptions  = shadowRoot?.querySelectorAll(".instant-apps-create__option");
+
+        expect(listOptions?.length).toBe(5);
+
+        const hrefs = listOptions && Array.from(listOptions).map((option: HTMLAnchorElement) => option.href);
+        const validSubstrings = ["instantgallery", "mapviewer", "stories", "template", "dashboards"];
+        
+        const testValidity = (testString: string, substrings: string[]) => 
```

Comment:
this look like the same test as the one above. Any reason why this is being tested more than once?

Reply:
test

Reply:
Define the function once, keep it DRY (Don't repeat yourself)